File: client/app/models/Coin.js
class Coin {
  constructor(object) {
    this.name = object.name;
    this.symbol = object.symbol;
    this.link = object.link;
    this.announcement_thread = object.announcement_thread;
    this.description = object.description;
    this.algorithm = object.algorithm;
    this.reward = object.reward;
    this.max_cap = object.max_cap;
    this.halving = object.halving;
    this.premined = object.premined;
    this.had_ico = object.had_ico;
    this.ethereum_token = object.ethereum_token;
    this.announcement_date = object.announcement_date;
    this.notable_facts = object.notable_facts;
    this.pros = object.pros;
    this.cons = object.cons;

    this.price_usd = object.price_usd;
    this.price_btc = object.price_btc;
    this.volume_usd_24h = object.volume_usd_24h;
    this.market_cap_usd = object.market_cap_usd;
    this.available_supply = object.available_supply;
    this.total_supply = object.total_supply;
    this.percent_change_1h = object.percent_change_1h;
    this.percent_change_24h = object.percent_change_24h;
    this.percent_change_7d = object.percent_change_7d;
    this.cmc_last_updated = object.cmc_last_updated;
  }
};

export default Coin;

File: app/models/coin.rb
class Coin < ApplicationRecord
  has_paper_trail
  after_create :fetch_market_data

  private

  def fetch_market_data
    client = Api::CoinMarketCap.new
    data = client.data_for(self.name)

    if data.any?
      self.price_usd = data[:price_usd].to_d
      self.price_btc = data[:price_btc].to_d
      self.volume_usd_24h = data["24h_volume_usd"].to_d
      self.market_cap_usd = data[:market_cap_usd].to_d
      self.available_supply = data[:available_supply].to_d
      self.total_supply = data[:total_supply].to_d
      self.percent_change_1h = data[:percent_change_1h].to_d
      self.percent_change_24h = data[:percent_change_24h].to_d
      self.percent_change_7d = data[:percent_change_7d].to_d
      self.cmc_last_updated = Time.at(data[:last_updated].to_i)

      self.save!
    end
  end

end
